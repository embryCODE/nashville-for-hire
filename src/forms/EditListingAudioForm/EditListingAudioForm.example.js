/* eslint-disable no-console */
import EditListingAudioForm from './EditListingAudioForm'

export const Empty = {
  component: EditListingAudioForm,
  props: {
    initialValues: { country: 'US', images: [] },
    stripeConnected: false,
    onImageUpload: (values) => {
      console.log(`onImageUpload with id (${values.id}) and file name (${values.file.name})`)
    },
    onSubmit: (values) => {
      console.log('Submit EditListingAudioForm with (unformatted) values:', values)
    },
    saveActionMsg: 'Save photos',
    updated: false,
    ready: false,
    updateInProgress: false,
    disabled: false,
    onUpdateImageOrder: (imageOrder) => {
      console.log('onUpdateImageOrder with new imageOrder:', imageOrder)
    },
    onRemoveImage: (imageId) => {
      console.log('remove image:', imageId)
    },
  },
  group: 'forms',
}
