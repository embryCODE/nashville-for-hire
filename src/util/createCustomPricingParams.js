import { Money } from 'sharetribe-flex-sdk/src/types'

export default function ({ listing, bookingData }) {
  const lineItems = Object.values(bookingData).map((d) => {
    // TODO: Handle when there is no price correctly
    const unitPrice = d.price ? new Money(d.price.amount, d.price.currency) : new Money(0, 'USD')

    return {
      code: 'line-item/' + d.code,
      unitPrice,
      quantity: d.quantity,
    }
  })

  return {
    listingId: listing.id,
    lineItems,
  }
}
