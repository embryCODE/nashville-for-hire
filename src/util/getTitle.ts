import { PublicData } from '../components/NFHCustom/types'

const getTitle = (code: string, publicData: PublicData) => {
  const pricesArray = Object.values(publicData.prices)
  const realCode = code.split('/')[1]
  const price = pricesArray.find((p) => p.code === realCode)

  if (!price) throw new Error()

  return price.title
}

export default getTitle
