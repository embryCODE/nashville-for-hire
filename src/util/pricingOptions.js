export default {
  femaleVocals: {
    option0: {
      code: 'femaleVocals0',
      description: 'Lead vocal, doubled lead, multiple harmony tracks.',
      title: 'Full Vocal Package',
    },
    option1: {
      code: 'femaleVocals1',
      description: 'Lead vocal only. No harmony.',
      title: 'Lead Vocal',
    },
    option2: {
      code: 'femaleVocals2',
      description: 'Lead vocal + 1 harmony track',
      title: 'Lead Vocal + Harmony',
    },
    option3: {
      code: 'femaleVocals3',
      description: 'Backing vocals only',
      title: 'Backing Vocals',
    },
    option4: {
      code: 'femaleVocals4',
      description: 'Commercial release',
      title: 'Commercial Release',
    },
  },
  maleVocals: {
    option0: {
      code: 'maleVocals0',
      description: 'Lead vocal, doubled lead, multiple harmony tracks.',
      title: 'Full Vocal Package',
    },
    option1: {
      code: 'maleVocals1',
      description: 'Lead vocal only. No harmony.',
      title: 'Lead Vocal',
    },
    option2: {
      code: 'maleVocals2',
      description: 'Lead Vocal + 1 harmony track',
      title: 'Lead Vocal + Harmony',
    },
    option3: {
      code: 'maleVocals3',
      description: 'Backing vocals only',
      title: 'Backing Vocals',
    },
    option4: {
      code: 'maleVocals4',
      description: 'Commercial release',
      title: 'Commercial Release',
    },
  },
  songwriting: {
    option0: {
      code: 'songwriting0',
      description: 'Full cowrite via Zoom call',
      title: 'Full Cowrite',
    },
    option1: {
      code: 'songwriting1',
      description: '30 minute songwriting coaching session',
      title: '30 Minute Songwriting Coaching',
    },
    option2: {
      code: 'songwriting2',
      description: '1 hour songwriting coaching session',
      title: '1 Hour Songwriting Coaching',
    },
    option3: {
      code: 'songwriting3',
      description: 'Help finish your song (no video call)',
      title: 'Help Finish Your Song',
    },
  },
  topline: {
    option0: {
      code: 'topline0',
      description: 'Write lyrics, melody, and sing',
      title: 'Full Topline',
    },
    option1: {
      code: 'topline1',
      description: 'Sing your melody and lyrics',
      title: 'Sing Your Melody',
    },
    option2: {
      code: 'topline2',
      description: 'Write lyrics and sing your melody',
      title: 'Write Lyrics and Sing Your Melody',
    },
    option3: {
      code: 'topline3',
      description: 'Write melody and sing your lyrics',
      title: 'Write Melody and Sing Your Lyrics',
    },
  },
  acousticGuitar: {
    option0: {
      code: 'acousticGuitar0',
      description: 'Up to 4 tracks',
      title: 'Acoustic Guitar Package',
    },
    option1: {
      code: 'acousticGuitar1',
      description: 'Up to 40 seconds',
      title: 'Acoustic Guitar Solo',
    },
    option2: {
      code: 'acousticGuitar2',
      description: 'Acoustic guitar solo plus harmony line. Up to 40 seconds.',
      title: 'Acoustic Guitar Solo + Harmony',
    },
  },
  electricGuitar: {
    option0: {
      code: 'electricGuitar0',
      description: 'Up to 4 tracks (doubled)',
      title: 'Electric Guitar Package',
    },
    option1: {
      code: 'electricGuitar1',
      description: 'Up to 40 seconds',
      title: 'Electric Guitar Solo',
    },
    option2: {
      code: 'electricGuitar2',
      description: 'Electric guitar solo plus harmony line. Up to 40 seconds.',
      title: 'Electric Guitar Solo + Harmony',
    },
  },
  bass: {
    option0: {
      code: 'bass0',
      description: 'Electric bass',
      title: 'Electric Bass',
    },
    option1: {
      code: 'bass1',
      description: 'Upright bass',
      title: 'Upright Bass',
    },
    option2: {
      code: 'bass2',
      description: 'Synth bass',
      title: 'Synth Bass',
    },
  },
  drums: {
    option0: {
      code: 'drums0',
      description: 'Drum kit',
      title: 'Drum Kit',
    },
    option1: {
      code: 'drums1',
      description: 'Add percussion',
      title: 'Percussion',
    },
  },
  piano: {
    option0: {
      code: 'piano0',
      description: 'Up to 3 piano tracks on a song',
      title: 'Piano Package',
    },
    option1: {
      code: 'piano1',
      description: '1 piano track on a song',
      title: '1 Piano Track',
    },
  },
  keys: {
    option0: {
      code: 'keys0',
      description: 'Up to 3 keys tracks',
      title: 'Keys Package',
    },
    option1: {
      code: 'keys1',
      description: '1 keys track',
      title: '1 Keys Track',
    },
  },
  organ: {
    option0: {
      code: 'organ0',
      description: 'Up to 3 organ tracks',
      title: 'Organ Package',
    },
    option1: {
      code: 'organ1',
      description: '1 organ track',
      title: '1 Organ Track',
    },
  },
  synth: {
    option0: {
      code: 'synth0',
      description: 'No beat programming. Up to 4 tracks.',
      title: 'Basic Synth Arrangement',
    },
    option1: {
      code: 'synth1',
      description: 'No beat programming. Up to 8 tracks.',
      title: 'Full Synth Arrangement',
    },
    option2: {
      code: 'synth2',
      description: 'Synth bass only',
      title: 'Synth Bass',
    },
    option3: {
      code: 'synth3',
      description: '1-2 synth tracks',
      title: '1-2 Synth Tracks',
    },
  },
  beatProgramming: {
    option0: {
      code: 'beatProgramming0',
      description: 'Basic beat package',
      title: 'Basic Beat Package',
    },
    option1: {
      code: 'beatProgramming1',
      description: 'Complex beat package',
      title: 'Complex Beat Package',
    },
  },
  cello: {
    option0: {
      code: 'cello0',
      description: 'Solo cello',
      title: 'Solo Cello',
    },
    option1: {
      code: 'cello1',
      description: 'Cello trio',
      title: 'Cello Trio',
    },
    option2: {
      code: 'cello2',
      description: 'Cello quartet',
      title: 'Cello Quartet',
    },
  },
  violin: {
    option0: {
      code: 'violin0',
      description: 'Solo violin',
      title: 'Solo Violin',
    },
    option1: {
      code: 'violin1',
      description: 'Violin trio',
      title: 'Violin Trio',
    },
    option2: {
      code: 'violin2',
      description: 'Violin quartet',
      title: 'Violin Quartet',
    },
  },
  fiddle: {
    option0: {
      code: 'fiddle0',
      description: 'Solo fiddle',
      title: 'Solo Fiddle',
    },
    option1: {
      code: 'fiddle1',
      description: 'Solo viola',
      title: 'Solo Viola',
    },
    option2: {
      code: 'fiddle2',
      description: 'Viola trio',
      title: 'Viola Trio',
    },
    option3: {
      code: 'fiddle3',
      description: 'Viola quartet',
      title: 'Viola Quartet',
    },
  },
  fullStringArrangement: {
    option0: {
      code: 'fullStringArrangement0',
      description: 'Full string arrangement for trio',
      title: 'Trio',
    },
    option1: {
      code: 'fullStringArrangement1',
      description: 'Full string arrangement for quartet',
      title: 'Quartet',
    },
    option2: {
      code: 'fullStringArrangement2',
      description: 'Full string arrangement for large string ensemble',
      title: 'Large String Arrangement',
    },
  },
  pedalSteel: {
    option0: {
      code: 'pedalSteel0',
      description: 'Up to 3 pedal steel tracks on a song',
      title: 'Pedal Steel Package',
    },
    option1: {
      code: 'pedalSteel1',
      description: '1 pedal steel track on a song',
      title: '1 Pedal Steel Track',
    },
  },
  banjo: {
    option0: {
      code: 'banjo0',
      description: 'Up to 3 banjo tracks on a song',
      title: 'Banjo Package',
    },
    option1: {
      code: 'banjo1',
      description: '1 banjo track on a song',
      title: '1 Banjo Track',
    },
  },
  dobro: {
    option0: {
      code: 'dobro0',
      description: 'Up to 3 dobro tracks on a song',
      title: 'Dobro Package',
    },
    option1: {
      code: 'dobro1',
      description: '1 Dobro Track on a Song',
      title: '1 Dobro Track on a Song',
    },
  },
  lapSteel: {
    option0: {
      code: 'lapSteel0',
      description: 'Up to 3 lap steel tracks on a song',
      title: 'Lap Steel Package',
    },
    option1: {
      code: 'lapSteel1',
      description: '1 lap steel track on a song',
      title: '1 Lap Steel Track',
    },
  },
  harmonica: {
    option0: {
      code: 'harmonica0',
      description: 'Solo harmonica',
      title: 'Solo Harmonica',
    },
  },
  trumpet: {
    option0: {
      code: 'trumpet0',
      description: 'Solo trumpet',
      title: 'Solo Trumpet',
    },
    option1: {
      code: 'trumpet1',
      description: 'Trumpet section',
      title: 'Trumpet Section',
    },
  },
  trombone: {
    option0: {
      code: 'trombone0',
      description: 'Solo trombone',
      title: 'Solo Trombone',
    },
    option1: {
      code: 'trombone1',
      description: 'Trombone section',
      title: 'Trombone Section',
    },
  },
  frenchHorn: {
    option0: {
      code: 'frenchHorn0',
      description: 'Solo French Horn',
      title: 'Solo French Horn',
    },
    option1: {
      code: 'frenchHorn1',
      description: 'French Horn section',
      title: 'French Horn Section',
    },
  },
  saxophone: {
    option0: {
      code: 'saxophone0',
      description: 'Solo sax',
      title: 'Solo Sax',
    },
    option1: {
      code: 'saxophone1',
      description: 'Sax section',
      title: 'Sax Section',
    },
  },
  hornSection: {
    option0: {
      code: 'hornSection0',
      description: '3 piece horn section',
      title: '3 Piece',
    },
    option1: {
      code: 'hornSection1',
      description: '5 piece horn section',
      title: '5 Piece',
    },
  },
  mixing: {
    option0: {
      code: 'mixing0',
      description: '1-16 tracks',
      title: '1-16 Tracks',
    },
    option1: {
      code: 'mixing1',
      description: '17-48 tracks',
      title: '17-48 Tracks',
    },
    option2: {
      code: 'mixing2',
      description: '49+ tracks',
      title: '49+ Tracks',
    },
    option3: {
      code: 'mixing3',
      description: '1-2 tracks',
      title: 'Vocal Tuning 1-2 Tracks',
    },
    option4: {
      code: 'mixing4',
      description: '3-5 tracks',
      title: 'Vocal Tuning 3-5 Tracks',
    },
    option5: {
      code: 'mixing5',
      description: '6-12 tracks',
      title: 'Vocal Tuning 6-12 Tracks',
    },
  },
  mastering: {
    option0: {
      code: 'mastering0',
      description: 'Master one song',
      title: 'Master One Song',
    },
  },
  produce: {
    option0: {
      code: 'produce0',
      description: '1-3 instruments',
      title: 'Simple Production',
    },
    option1: {
      code: 'produce1',
      description: 'Full band or programmed',
      title: 'Full Production',
    },
    option2: {
      code: 'produce2',
      description: '1-2 tracks',
      title: 'Vocal Tuning 1-2 Tracks',
    },
    option3: {
      code: 'produce3',
      description: '3-5 tracks',
      title: 'Vocal Tuning 3-5 Tracks',
    },
    option4: {
      code: 'produce4',
      description: '6-12 tracks',
      title: 'Vocal Tuning 6-12 Tracks',
    },
    option5: {
      code: 'produce5',
      description: 'Add mixing to simple production',
      title: 'Simple mix',
    },
    option6: {
      code: 'produce6',
      description: 'Add mixing to full production',
      title: 'Full mix',
    },
  },
}
