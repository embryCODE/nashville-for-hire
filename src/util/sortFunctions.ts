import { Price } from '../components/NFHCustom/types'

export const sortByPriceCode = (a: Price, b: Price) => {
  if (a.code.includes('custom')) return 1
  if (b.code.includes('custom')) return -1
  return 0
}

export const sortByPricesAndQuantityCode = ([, a]: [string, Price], [, b]: [string, Price]) => {
  if (a.code.includes('custom')) return 1
  if (b.code.includes('custom')) return -1
  return 0
}
