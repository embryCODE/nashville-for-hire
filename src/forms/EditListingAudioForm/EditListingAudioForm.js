import React, { Component } from 'react'
import { array, bool, func, shape, string } from 'prop-types'
import { compose } from 'redux'
import { Form as FinalForm } from 'react-final-form'
import { FormattedMessage, intlShape, injectIntl } from '../../util/reactIntl'
import classNames from 'classnames'
import { propTypes } from '../../util/types'
import { Button, FieldTextInput, Form } from '../../components'
import ReactS3Uploader from 'react-s3-uploader'
import css from './EditListingAudioForm.css'
import { s3AudioBucketName, s3UrlSigningServer, s3UrlSigningUrl } from '../../config'
import styled from 'styled-components/macro'

const ACCEPT_AUDIO = 'audio/*'

const Card = styled.div`
  background: #fff;
  border-radius: 4px;

  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);

  width: 100%;
  padding: 1rem;

  margin-bottom: 1rem;

  h3 {
    margin-top: 0;
  }
`

export class EditListingAudioFormComponent extends Component {
  constructor(props) {
    super(props)
    this.state = {
      currentFileName: '',
      listingId: props.listingId,
      audio: props.audio,
      isLoading: false,
    }
    this.uploader = null
    this.handleUpload = this.handleUpload.bind(this)
  }

  handleUpload(fileName) {
    this.setState((prevState) => ({
      audio: [...prevState.audio, { fileName }],
    }))
  }

  render() {
    return (
      <FinalForm
        {...this.props}
        initialValues={this.props.audio.reduce((acc, curr, i) => {
          return { ...acc, [`audioObject-${i}`]: curr }
        }, {})}
        render={(formRenderProps) => {
          const {
            className,
            fetchErrors,
            saveActionMsg,
            handleSubmit,
            valid,
            updated,
            pristine,
            ready,
            updateInProgress,
          } = formRenderProps
          const { updateListingError } = fetchErrors || {}
          const classes = classNames(css.root, className)

          const namespace = 'listing-id-' + this.state.listingId.uuid
          const tags = `daysUntilExpiration=0`
          const required = (value) => (value ? undefined : 'Required')
          const submitReady = (updated && pristine) || ready
          const inProgress = updateInProgress || this.state.isLoading

          return (
            <Form className={classes} onSubmit={handleSubmit}>
              {updateListingError ? (
                <p className={css.error}>
                  <FormattedMessage id="EditListingAudioForm.updateFailed" />
                </p>
              ) : null}

              <div>
                {this.state.audio.map(({ fileName }, index) => (
                  <Card key={fileName + index}>
                    <div>
                      <h3>{fileName.split('/')[1]}</h3>

                      <FieldTextInput
                        id="fileName"
                        type="hidden"
                        name={`audioObject-${index}.fileName`}
                        initialValue={fileName}
                        validate={required}
                      />

                      <FieldTextInput
                        id="name"
                        type="text"
                        name={`audioObject-${index}.name`}
                        label="Sample Name"
                        placeholder="Bluesy Jams"
                        validate={required}
                      />

                      <FieldTextInput
                        id="description"
                        type="textarea"
                        name={`audioObject-${index}.description`}
                        label="What was your involvement on this track?"
                        placeholder="I played the guitar and sang harmonies"
                        validate={required}
                      />
                    </div>
                  </Card>
                ))}
              </div>

              <h3>Add audio</h3>

              <ReactS3Uploader
                signingUrl={s3UrlSigningUrl}
                signingUrlQueryParams={{
                  bucketName: s3AudioBucketName,
                  namespace,
                }}
                signingUrlMethod="GET"
                accept={ACCEPT_AUDIO}
                onSignedUrl={(res) => {
                  this.handleUpload(res.filename)
                }}
                onProgress={() => {
                  this.setState({ isLoading: true })
                }}
                onError={(err) => {
                  console.error({ onError: err })
                  this.setState({ isLoading: false })
                }}
                onFinish={() => {
                  this.setState({ isLoading: false })
                }}
                uploadRequestHeaders={{ 'x-amz-acl': 'public-read', 'x-amz-tagging': tags }}
                contentDisposition="auto"
                scrubFilename={(filename) => filename.replace(/[^\w\d_\-.]+/gi, '')}
                server={s3UrlSigningServer}
                autoUpload={true}
                // preprocess={(thing)=> console.log({thing})} // keeping just in case
                // inputRef={}
              />

              <p className={css.tip}>
                <FormattedMessage id="EditListingAudioForm.addAudioTip" />
              </p>

              <Button
                className={css.submitButton}
                type="submit"
                inProgress={inProgress}
                disabled={!valid}
                ready={submitReady}
              >
                {saveActionMsg}
              </Button>
            </Form>
          )
        }}
      />
    )
  }
}

EditListingAudioFormComponent.defaultProps = { fetchErrors: null, audio: [] }

EditListingAudioFormComponent.propTypes = {
  fetchErrors: shape({ showListingsError: propTypes.error, updateListingError: propTypes.error }),
  audio: array,
  intl: intlShape.isRequired,
  onSubmit: func.isRequired,
  saveActionMsg: string.isRequired,
  disabled: bool.isRequired,
  ready: bool.isRequired,
  updated: bool.isRequired,
  updateInProgress: bool.isRequired,
}

export default compose(injectIntl)(EditListingAudioFormComponent)
