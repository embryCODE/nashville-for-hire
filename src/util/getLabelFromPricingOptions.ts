import pricingOptions from './pricingOptions'

const unnestedPricingOptions = Object.values(pricingOptions)
  .flat()
  .map((po) => Object.values(po).flat())
  .flat()

const getLabelFromPricingOptions = (code: string): string => {
  const pricingOption = unnestedPricingOptions.find((po) => code.includes(po.code))
  return pricingOption ? pricingOption.label : ''
}

export default getLabelFromPricingOptions
