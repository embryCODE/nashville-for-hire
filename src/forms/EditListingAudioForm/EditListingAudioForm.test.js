import React from 'react'
import { renderShallow } from '../../util/test-helpers'
import { fakeIntl } from '../../util/test-data'
import { EditListingAudioFormComponent } from './EditListingAudioForm'

const noop = () => null

describe('EditListingAudioForm', () => {
  it('matches snapshot', () => {
    const tree = renderShallow(
      <EditListingAudioFormComponent
        initialValues={{ country: 'US', images: [] }}
        intl={fakeIntl}
        dispatch={noop}
        onImageUpload={(v) => v}
        onSubmit={(v) => v}
        saveActionMsg="Save photos"
        onUpdateImageOrder={(v) => v}
        stripeConnected={false}
        updated={false}
        ready={false}
        updateInProgress={false}
        disabled={false}
        ready={false}
        onRemoveImage={noop}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
