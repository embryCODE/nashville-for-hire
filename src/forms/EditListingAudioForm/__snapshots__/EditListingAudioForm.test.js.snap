// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditListingAudioForm matches snapshot 1`] = `
<ReactFinalForm
  audio={Array []}
  disabled={false}
  dispatch={[Function]}
  fetchErrors={null}
  initialValues={
    Object {
      "country": "US",
      "images": Array [],
    }
  }
  intl={
    Object {
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "now": [Function],
    }
  }
  onImageUpload={[Function]}
  onRemoveImage={[Function]}
  onSubmit={[Function]}
  onUpdateImageOrder={[Function]}
  ready={false}
  render={[Function]}
  saveActionMsg="Save photos"
  stripeConnected={false}
  updateInProgress={false}
  updated={false}
/>
`;
